src/_igraph/graphobject.c

```diff
@@ -8033,19 +8033,20 @@ PyObject *igraphmodule_Graph_layout_umap(
     igraphmodule_GraphObject * self, PyObject * args, PyObject * kwds)
 {
   static char *kwlist[] =
-    { "dist", "dim", "seed", "min_dist", "epochs", NULL };
+    { "dist", "weights", "dim", "seed", "min_dist", "epochs", NULL };
   igraph_matrix_t m;
   igraph_vector_t *dist = 0;
   Py_ssize_t dim = 2;
   double min_dist = 0.01;
   Py_ssize_t epochs = 500;
-  PyObject *dist_o = Py_None;
+  PyObject *dist_o = Py_None, *weights_o = Py_None;
   PyObject *seed_o = Py_None;
   PyObject *result_o;
   igraph_bool_t use_seed = 0;
+  igraph_bool_t distances_are_weights = 0;
```

Comment:
I'll try to fix most instances of these in a separate commit / PR. There are quite a few instances of these and it's relatively easy to grep for `igraph_bool_t.*[01]`.

Reply:
Actually, we cannot use `true` and `false`. This would give an `error: use of undeclared identifier 'false'` error with Python's default compiler flags for C extensions (and it's usually not a good idea to override them explicitly).

Reply:
Why would it give an error? I don't understand why the presence of `true` / `false` depends on compiler flags. They are defined in `stdbool.h`, which is included unconditionally by igraph.  This header is new in C99, but igraph requires C99, even for just including its headers.

Reply:
Forgot to include `stdbool.h`, my bad.